slick.groupitemmetadataprovider.js

```diff
@@ -42,8 +42,20 @@
       includeHeaderTotals: false
     };
 
-    options = $.extend(true, {}, _defaults, options);
+    var options = {};
 
+    setOptions(_options);
+
+    function getOptions(){
+      return options;
+    }
+
+    function setOptions(_options)
+    {
+      options = $.extend(true, {}, _defaults, options, _options);
+
+      return this;
```

Comment:
maybe remove the empty space and why are you returning `this` ? For the `this` we typically add a variable in the class as `var _self = this;` on top. Why not return the options instead of this?

Reply:
returning `this` would allow function chaining, but as the other functions of this class do not allow function chaining, I will remove the line altogether if that's okay. the function won't return anything.

Reply:
ah ok yeah, function chaining is nice but we rarely use it, probably won't be used much in this context anyway so yeah removing it is probably better

Reply:
return value removed.